addons/hr_work_entry_holidays/tests/test_multi_contract.py

```diff
@@ -169,3 +169,105 @@ def test_leave_multi_contracts_split(self):
+                ]
+            },
+        ])
+        employee = self.env['hr.employee'].create({
+            'name': 'Employee',
+            'resource_calendar_id': calendar_partial.id,
+        })
+        self.env['hr.contract'].create([
+            {
+                'name': 'Full time (5/5)',
+                'employee_id': employee.id,
+                'date_start': datetime.strptime('2023-01-01', '%Y-%m-%d').date(),
+                'date_end': datetime.strptime('2023-06-30', '%Y-%m-%d').date(),
+                'resource_calendar_id': calendar_full.id,
+                'wage': 1000.0,
+                'state': 'close', # Old contract
+                'date_generated_from': datetime.strptime('2023-01-01', '%Y-%m-%d'),
+                'date_generated_to': datetime.strptime('2023-01-01', '%Y-%m-%d'),
+            },
+            {
+                'name': 'Partial time (4/5)',
+                'employee_id': employee.id,
+                'date_start': datetime.strptime('2023-07-01', '%Y-%m-%d').date(),
+                'date_end': datetime.strptime('2023-12-31', '%Y-%m-%d').date(),
+                'resource_calendar_id': calendar_partial.id,
+                'wage': 1000.0,
+                'state': 'open', # Current contract
+                'date_generated_from': datetime.strptime('2023-07-01', '%Y-%m-%d'),
+                'date_generated_to': datetime.strptime('2023-07-01', '%Y-%m-%d'),
+            },
+        ])
+        leave_type = self.env['hr.leave.type'].create({
+            'name': 'Leave Type',
+            'time_type': 'leave',
+            'requires_allocation': 'yes',
+            'leave_validation_type': 'hr',
+            'request_unit': 'day',
+        })
+        self.env['hr.leave.allocation'].create({
+            'name': 'Allocation',
+            'employee_id': employee.id,
+            'holiday_status_id': leave_type.id,
+            'number_of_days': 10,
+            'state': 'confirm',
+            'date_from': datetime.strptime('2023-01-01', '%Y-%m-%d').date(),
+            'date_to': datetime.strptime('2023-12-31', '%Y-%m-%d').date(),
+        }).action_validate()
+        leave_during_full_time, leave_during_partial_time = self.env['hr.leave'].create([
+            {
+                'employee_id': employee.id,
+                'holiday_status_id': leave_type.id,
+                'number_of_days': 3,
+                'date_from': datetime.combine(date(2023, 1, 3), time.min), # Tuesday
+                'date_to': datetime.combine(date(2023, 1, 5), time.max), # Thursday
+            },
+            {
+                'employee_id': employee.id,
+                'holiday_status_id': leave_type.id,
+                'number_of_days': 2,
+                'date_from': datetime.combine(date(2023, 12, 5), time.min), # Tuesday
+                'date_to': datetime.combine(date(2023, 12, 7), time.max), # Thursday
+            },
```

Comment:
Should we enforce the number of days if the goal is to have them computed correctly?

Reply:
@Bertrand2 When programmatically creating time offs like this it is currently indeed still necessary to 'hardcode' the number of days. This is one of the things my ongoing refactor will fix :)